editor/editor_node.cpp

```diff
@@ -3492,6 +3492,7 @@ void EditorNode::add_extension_editor_plugin(const StringName &p_class_name) {
 	EditorPlugin *plugin = Object::cast_to<EditorPlugin>(ClassDB::instantiate(p_class_name));
 	singleton->editor_data.add_extension_editor_plugin(p_class_name, plugin);
 	add_editor_plugin(plugin);
+	plugin->enable_plugin();
```

Comment:
As @kisg points out, we could alternatively implement this by doing:

```suggestion
	add_editor_plugin(plugin, true);
```

If we do that, though, we'd probably want to change the name of that argument, since the name is currently `p_config_changed`, which doesn't make sense in this context.

Reply:
Yeah, I was thinking that, even though `p_config_changed` is currently only used to determine if `enable_plugin()` is called, it may be used for more stuff in the future and I didn't want to change its meaning. But I'm happy to change it if we're fine with that.

Reply:
The current logic for `p_config_changed` seems to be a toggle for on/off:
```
void EditorNode::set_addon_plugin_enabled(const String &p_addon, bool p_enabled, bool p_config_changed) {
...
	if (!p_enabled) {
		EditorPlugin *addon = addon_name_to_plugin[addon_path];
		remove_editor_plugin(addon, p_config_changed);
...
		return;
	}
...
add_editor_plugin(ep, p_config_changed);
```

`add_editor_plugin` enables the plugin if `p_config_changed`, and `remove_editor_plugin` disables the plugin if `p_config_changed`.

So I think we can rely on that behavior, but I'd rename the bool in `add_editor_plugin` and `remove_editor_plugin` respectively to `enable_plugin` and `disable_plugin`.

Reply:
CC @KoBeWi

Reply:
One can add a plugin but not enable it (see the _Activate now_ checkbox when creating GDScript plugins). You can also always enable/disable plugins from project settings.

GDExtension and module plugins don't really allow enabling/disabling plugins[^1], I think it's implied that these plugins are always enabled when added and disabled when removed. But before this PR the `_enable_plugin` and `_disable_plugin` virtual methods were never called.

The motivation behind this PR was to maintain consistency with the current scripting behavior in GDExtension plugins. This is important to me since I want to avoid behavior changes for the .NET bindings when moving from scripting to GDExtension.

I agree about removing without disabling being confusing, I'd assume removing always imply disabling. I forgot about disabling in this PR though, because the .NET plugin I'm working on doesn't override `_disable_plugin` so I didn't notice.

[^1]: This may change in the future for GDExtensions, since the GDExtension team wants to allow enabling/disabling GDExtensions in project settings.

Reply:
Enabling/disabling plugins only happens when a plugin is created/toggled in Project Settings. When you open a project, the plugins listed in `project.godot` are added, but the enabled callback is not called. You can't enable a plugin before it is added to EditorNode's plugin list and you shouldn't be able to remove it before it's disabled.

Reply:
> When you open a project, the plugins listed in `project.godot` are added, but the enabled callback is not called.

Oh, then this PR is wrong.

Reply:
So should this be reverted then?

Reply:
Yeah, it's the same. If we don't want GDExtension EditorPlugins to have `_enable_plugin` called when Godot is launched, then we should revert the PR. But that means `_enable_plugin` will never be called for GDExtension EditorPlugins since we have no way of _enabling_ them (I guess they are considered to be pre-enabled when they are added).

Reply:
There is enter tree callback called on editor launch, so this use-case is covered. `_enable_plugin` is for when user enables a plugin, and if GDExtensions can't be toggled, this callback is not for them.